src/syntax.rs

```diff
@@ -153,153 +151,4 @@ impl<F: LurkField> Store<F> {
             }
         }
     }
-
-    /// Tries to fetch a syntactic list from an expression pointer, by looping over cons cells and
-    /// collecting their contents. If the ptr does not point to a cons or nil (i.e. not a list) we
-    /// return None. If after traversing zero or more cons cells we hit a `nil`, we return a proper
-    /// list (`Syntax::List`), otherwise an improper list (`Syntax::Improper`). If the proper list
-    /// is a quotation `(quote x)`, then we return the syntactic quotation `Syntax::Quote`
-    #[allow(dead_code)]
-    fn fetch_syntax_list(&self, mut ptr: Ptr<F>) -> Option<Syntax<F>> {
-        let mut list = vec![];
-        loop {
-            match self.fetch(&ptr)? {
-                Expression::Cons(car, cdr) => {
-                    list.push(self.fetch_syntax(car)?);
-                    ptr = cdr;
-                }
-                Expression::Nil => {
-                    return Some(Syntax::List(Pos::No, list));
-                }
-                _ => {
-                    if list.is_empty() {
-                        return None;
-                    } else {
-                        let end = Box::new(self.fetch_syntax(ptr)?);
-                        return Some(Syntax::Improper(Pos::No, list, end));
-                    }
-                }
-            }
-        }
-    }
-
-    fn fetch_syntax(&self, ptr: Ptr<F>) -> Option<Syntax<F>> {
-        match ptr.tag {
-            ExprTag::Num => Some(Syntax::Num(Pos::No, *self.fetch_num(&ptr)?)),
-            ExprTag::Char => Some(Syntax::Char(Pos::No, self.fetch_char(&ptr)?)),
-            ExprTag::U64 => Some(Syntax::UInt(Pos::No, self.fetch_uint(&ptr)?)),
-            ExprTag::Str => Some(Syntax::String(Pos::No, self.fetch_string(&ptr)?)),
-            ExprTag::Nil => Some(Syntax::Symbol(Pos::No, lurk_sym("nil").into())),
-            ExprTag::Cons => self.fetch_syntax_list(ptr),
-            ExprTag::Sym => Some(Syntax::Symbol(Pos::No, self.fetch_sym(&ptr)?.into())),
-            ExprTag::Key => Some(Syntax::Symbol(Pos::No, self.fetch_key(&ptr)?.into())),
-            _ => None,
-        }
-    }
-}
-
-#[cfg(test)]
-mod test {
-    use super::*;
-    use crate::Symbol;
-    use pasta_curves::pallas::Scalar as Fr;
-
-    #[test]
-    fn display_syntax() {
```

Comment:
This test is trivialized by
https://github.com/lurk-lab/lurk-rs/blob/3930e3d96c736d9db6bae7ec348f9ccb87bf56b4/src/parser/syntax.rs#L1003-L1010